source/csharp/Camera/Basic/CaptureViaGenICam/CaptureViaGenICam.cs

```diff
@@ -0,0 +1,128 @@
+using System;
+using Duration = Zivid.NET.Duration;
+
+class Program
+{
+    static void Main()
+    {
+        try
+        {
+            Console.WriteLine("Connecting to camera");
+            var zividDevice = GetFirstAvailableZividDevice();
+            var framegrabber = new HalconDotNet.HTuple();
+            HalconDotNet.HOperatorSet.OpenFramegrabber("GenICamTL",
+                                    1,
+                                    1,
+                                    0,
+                                    0,
+                                    0,
+                                    0,
+                                    "progressive",
+                                    -1,
+                                    "default",
+                                    -1,
+                                    "false",
+                                    "default",
+                                    zividDevice,
+                                    0,
+                                    0,
+                                    out framegrabber);
+
+            Console.WriteLine("Configuring 3D-settings");
+            HalconDotNet.HOperatorSet.SetFramegrabberParam(framegrabber, "create_objectmodel3d", "enable");
+            HalconDotNet.HOperatorSet.SetFramegrabberParam(framegrabber, "add_objectmodel3d_overlay_attrib", "enable");
+            HalconDotNet.HOperatorSet.SetFramegrabberParam(framegrabber, "AcquisitionMode", "SingleFrame");
+
+            Console.WriteLine("Configuring camera settings");
+            HalconDotNet.HOperatorSet.SetFramegrabberParam(framegrabber, "Aperture", 5.66);
+            HalconDotNet.HOperatorSet.SetFramegrabberParam(framegrabber, "ExposureTime", 8333);
+            HalconDotNet.HOperatorSet.SetFramegrabberParam(framegrabber, "Gain", 1);
+            HalconDotNet.HOperatorSet.SetFramegrabberParam(framegrabber, "ProcessingFiltersOutlierRemovalEnabled", 1);
+            HalconDotNet.HOperatorSet.SetFramegrabberParam(framegrabber, "ProcessingFiltersOutlierRemovalThreshold", 5);
+
+            Console.WriteLine("Capturing frame");
+            var region = new HalconDotNet.HObject();
+            var contours = new HalconDotNet.HObject();
+            var data = new HalconDotNet.HTuple();
+            var frame = new HalconDotNet.HObject();
```

Comment:
Maybe keep the same order as input arguments for the next function?

Reply:
done